### ISSUE TITLE
Gameovers may occur when moves are available

### ISSUE THREAD

**Issue by PistolRcks** [paulpage/py2048#12]
The `hasLost()` function does not check if there are available moves, so there are some situations in which a move may be performed, yet the game believes you've lost.

**Comment by paulpage** [comment]
I'm not able to reproduce this, and it looks like hasLost() checks for available moves.

```
def hasLost():
    for tile in tiles:
        if ( movable(tile, 0,-1) or combinable(tile, 0,-1) or
             movable(tile, 0, 1) or combinable(tile, 0, 1) or
               movable(tile,-1, 0) or combinable(tile,-1, 0) or
               movable(tile, 1, 0) or combinable(tile, 1, 0)  ):
            return False
    return True
```

This is the code you're looking at, right? That if statement is meant to check every tile to make sure it isn't movable or combinable, and return true only if none of them are.

Does this only happen in specific circumstances? If so, could you post a series of putTile() calls that sets up the board so that it will happen so I can try to reproduce it in my environment?

**Comment by PistolRcks** [comment]
`hasLost()` is correct, but it seems like you can make a frame-perfect input on your last move and the tile will animate over another one. This is probably the issue in #10, and could probably be fixed with moving around the position of `animate()` in the main loop.

**Comment by PistolRcks** [comment]
Seems to be fixed as of #16.
